### PR TITLE
[pigeon] Adds Swift tests in CI script

### PR DESCRIPTION
**Pull request by ailtonvivaz**
This PR integrates the Swift tests in build script (`run_tests.sh`)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by tarrinneal — COMMENTED** [review]
Minor nit, looks good otherwise

File: packages/pigeon/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 4.2.0
+
+Adds Swift tests in CI script
```
[the remark this answers is not in the document]
Comment (ailtonvivaz) [reply packages/pigeon/CHANGELOG.md]: Fixed

**Review by tarrinneal — APPROVED** [review]

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for following up on this! The run_test changes looks great, and I see it running in the CI output now.

run_tests.sh only affects CI and local development, so doesn't need to be versioned (or in this case even mentioned in the changelog); you can revert the changes to all of the files other than run_tests.sh

**Comment by stuartmorgan** [comment]
Oops, looks like I taught the tooling about `tool/` being version-exempt, but not `run_test.sh`, so this will need an override. I'll file an issue for that.

**Pull request merged**